### ISSUE TITLE
Running on an Asus Ascent GX10

### ISSUE THREAD

**Issue by mwolf-pi3g** [devnen/Chatterbox-TTS-Server#134]
I have a new Asus box which is essentially a DGX Spark.
I freshly cloned the repo and ran:
docker compose -f docker-compose-cu128.yml up -d --build

The build and model download went fine.
Trying to do tts via the UI logged the following error.

Thanks in advance.

$ docker logs b3ea9e2ff877

==========
== CUDA ==
==========

CUDA Version 12.8.1

Container image Copyright (c) 2016-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

A copy of this license is made available in this container at /NGC-DL-CONTAINER-LICENSE for your convenience.

2026-04-09 13:14:41 [INFO] __main__: Starting TTS Server directly on http://0.0.0.0:8004
2026-04-09 13:14:41 [INFO] __main__: API documentation will be available at http://0.0.0.0:8004/docs
2026-04-09 13:14:41 [INFO] __main__: Web UI will be available at http://0.0.0.0:8004/
INFO:     Started server process [1]
INFO:     Waiting for application startup.
2026-04-09 13:14:41 [INFO] server: TTS Server: Initializing application...
2026-04-09 13:14:41 [INFO] server: Configuration loaded. Log file at: /app/logs/tts_server.log
/usr/local/lib/python3.10/dist-packages/torch/cuda/__init__.py:283: UserWarning: 
    Found GPU0 NVIDIA GB10 which is of cuda capability 12.1.
    Minimum and Maximum cuda capability supported by this version of PyTorch is
    (8.0) - (12.0)
    
  warnings.warn(
2026-04-09 13:14:41 [INFO] engine: CUDA requested and functional. Using CUDA.
2026-04-09 13:14:41 [INFO] engine: Final device selection: cuda
2026-04-09 13:14:41 [INFO] engine: Model selector from config: 'chatterbox-multilingual'
2026-04-09 13:14:41 [INFO] engine: Model selector 'chatterbox-multilingual' resolved to Multilingual model (ChatterboxMultilingualTTS)
2026-04-09 13:14:41 [INFO] engine: Initializing ChatterboxMultilingualTTS on device 'cuda'...
2026-04-09 13:14:41 [INFO] engine: Model type: multilingual
Fetching 6 files: 100%|██████████| 6/6 [00:00<00:00, 26886.56it/s]
/usr/local/lib/python3.10/dist-packages/diffusers/models/lora.py:393: FutureWarning: LoRACompatibleLinear is deprecated and will be removed in version 1.0.0. Use of LoRACompatibleLinear is deprecated. Please switch to PEFT backend by installing PEFT: pip install peft.
  deprecate("LoRACompatibleLinear", "1.0.0", deprecation_message)
2026-04-09 13:14:52 [INFO] root: input frame rate=25
2026-04-09 13:14:53 [WARNING] chatterbox.models.tokenizers.tokenizer: pkuseg not available - Chinese segmentation will be skipped
loaded PerthNet (Implicit) at step 250,000
2026-04-09 13:14:53 [INFO] engine: Successfully loaded ChatterboxMultilingualTTS on cuda
2026-04-09 13:14:53 [INFO] engine: Model sample rate: 24000 Hz
2026-04-09 13:14:53 [INFO] engine: TTS Model loaded successfully on cuda. Engine sample rate: 24000 Hz.
2026-04-09 13:14:53 [INFO] server: TTS Model loaded successfully via engine.
2026-04-09 13:14:53 [INFO] server: Application startup sequence complete.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8004 (Press CTRL+C to quit)
2026-04-09 13:14:54 [INFO] server: Attempting to open web browser to: http://localhost:8004/
2026-04-09 13:15:06 [INFO] server: Request received for main UI page ('/').
2026-04-09 13:15:06 [INFO] server: Request received for /api/ui/initial-data.
2026-04-09 13:15:06 [INFO] utils: Found 28 predefined voices in /app/voices
2026-04-09 13:15:09 [INFO] server: Received /tts request: mode='predefined', format='wav'
2026-04-09 13:15:09 [INFO] server: Using predefined voice: Emily.wav
2026-04-09 13:15:09 [INFO] server: Processing text as a single chunk (splitting not enabled or text too short).
2026-04-09 13:15:09 [INFO] server: Synthesizing chunk 1/1...
2026-04-09 13:15:09 [INFO] engine: Applying user-provided seed for generation: 3000
2026-04-09 13:15:09 [INFO] engine: Global seed set to: 3000
2026-04-09 13:15:13 [ERROR] engine: Error during TTS synthesis: 
  #ifdef __HIPCC__
  #define ERROR_UNSUPPORTED_CAST ;
  // corresponds to aten/src/ATen/native/cuda/thread_constants.h
  #define CUDA_OR_ROCM_NUM_THREADS 256
  // corresponds to aten/src/ATen/cuda/detail/OffsetCalculator.cuh
  #define MAX_DIMS 16
  #ifndef __forceinline__
  #define __forceinline__ inline __attribute__((always_inline))
  #endif
  #else
  //TODO use _assert_fail, because assert is disabled in non-debug builds
  #define ERROR_UNSUPPORTED_CAST assert(false);
  #define CUDA_OR_ROCM_NUM_THREADS 128
  #define MAX_DIMS 25
  #endif
  #define POS_INFINITY __int_as_float(0x7f800000)
  #define INFINITY POS_INFINITY
  #define NEG_INFINITY __int_as_float(0xff800000)
  #define NAN __int_as_float(0x7fffffff)

  typedef long long int int64_t;
  typedef unsigned int uint32_t;
  typedef signed char int8_t;
  typedef unsigned char uint8_t;  // NOTE: this MUST be "unsigned char"! "char" is equivalent to "signed char"
  typedef short int16_t;
  static_assert(sizeof(int64_t) == 8, "expected size does not match");
  static_assert(sizeof(uint32_t) == 4, "expected size does not match");
  static_assert(sizeof(int8_t) == 1, "expected size does not match");
  constexpr int num_threads = CUDA_OR_ROCM_NUM_THREADS;
  constexpr int thread_work_size = 8; // TODO: make template substitution once we decide where those vars live
  constexpr int block_work_size = thread_work_size * num_threads;

  
  
  namespace std {
  
  template <class _Tp>
  _Tp&& __declval(int);
  template <class _Tp>
  _Tp __declval(long);
  template <class _Tp>
  decltype(__declval<_Tp>(0)) declval() noexcept;

  SNIP

       using vec_t_output = aligned_vector<std::complex<float>, vec_size>;
        vec_t_output* to_0 = reinterpret_cast<vec_t_output*>(data[0]) + block_work_size / vec_size * idx;
        
        int thread_idx = threadIdx.x;
        #pragma unroll
        for (int i = 0; i<loop_size; i++){
          vec_t_output v;
          #pragma unroll
          for (int j=0; j<vec_size; j++){
          v.val[j] = out0[vec_size * i + j];
          }
          to_0[thread_idx] = v;
          
          thread_idx += num_threads;
        }
      }
  }
nvrtc: error: invalid value for --gpu-architecture (-arch)

2026-04-09 13:15:13 [ERROR] server: TTS engine failed to synthesize audio for chunk 1.

**for context:**

`$ docker exec -it b3ea9e2ff877 python3 -c "import torch; print(torch.cuda.get_device_name(0)); print(torch.version.cuda)"
/usr/local/lib/python3.10/dist-packages/torch/cuda/__init__.py:283: UserWarning: 
    Found GPU0 NVIDIA GB10 which is of cuda capability 12.1.
    Minimum and Maximum cuda capability supported by this version of PyTorch is
    (8.0) - (12.0)
    
  warnings.warn(
NVIDIA GB10
12.8
`

**Comment by osos** [comment]
Switching to Cuda 13.0 (cu130) and the for that torch 2.10 made it work for me on the dgx spark

**Comment by osos** [comment]
made a PR: https://github.com/devnen/Chatterbox-TTS-Server/pull/135 with the suggested changes. Works here.

**Comment by devnen** [comment]
Closing, support for the DGX Spark / GB10 (sm_121) is in main via #135 (thank you @osos for the fix). The path uses `requirements-nvidia-cu130.txt` with torch 2.10 / CUDA 13.0. Please reopen if you still hit it on a clean install.
